### PR TITLE
Reduce the number of replica ports of a legacy volume to 10

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#6079

### CODE REVIEW

File: engineapi/instance_manager.go [hunk 1]
```diff
@@ -474,13 +475,18 @@ func (c *InstanceManagerClient) ReplicaInstanceCreate(req *ReplicaInstanceCreate
 		return parseProcess(imapi.RPCToProcess(process)), nil
 	}
 
+	portCount := DefaultReplicaPortCount
+	if req.Replica.Spec.BackendStoreDriver == longhorn.BackendStoreDriverTypeSPDK {
+		portCount = spdktypes.DefaultReplicaReservedPortCount
+	}
```
Comment (shuo-wu) [inline engineapi/instance_manager.go:481]: For SPDK replica, the current reserved port count of each replica is 5: 
https://github.com/longhorn/longhorn-spdk-engine/blob/d326caa4884afed4930bde2108b3938b4f0ddd83/pkg/types/types.go#L16

Comment (derekbit) [reply engineapi/instance_manager.go:481]: Yes
``
portCount = spdktypes.DefaultReplicaReservedPortCount
```

Comment (shuo-wu) [reply engineapi/instance_manager.go:481]: Oh, sorry. You are using different default value here.

**Review by shuo-wu — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mergify backport v1.5.x

**Pull request merged**